contribute/architecture/frontend-data-requests.md

```diff
@@ -0,0 +1,51 @@
+# Data requests
+
+[BackendSrv is the core data request service](https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv) used by Grafana. This documentation highlights the high level concepts used by `BackendSrv`.
+
+## Request `cancellation`
+> **Note:** Data sources can implement their own cancellation concept. This documentation describes the `request cancellation` included in Grafana.
+
+### The challenge
+A data request can take a long time to finish. Between the request starts and finishes, the user might change context. For instance, the user navigates away or the user issues the same request again.
+
+If we allow ongoing requests to continue, it might cause unnecessary load on data sources.
```

Comment:
```suggestion
If we wait for canceled requests to complete, it might cause unnecessary load on data sources.
```

I tried to use simpler language here. Feel free to change it if inaccurate.

Reply:
Love it ❤️